meson.build

```diff
@@ -0,0 +1,197 @@
+  'src/fp/fp_mul_redc1.c',
+  'src/fp/fp_mul_redc1.h',
+  'src/fp/fp_pow.c',
+  'src/fp/fp_rand.c',
+  'src/fp/fp_sqrt.c',
+  )
+
+curves_mod_src = files(
+  'src/curves/aff_pt.c',
+  'src/curves/aff_pt_montgomery.c',
+  'src/curves/ec_edwards.c',
+  'src/curves/ec_montgomery.c',
+  'src/curves/ec_params.c',
+  'src/curves/ec_shortw.c',
+  'src/curves/aff_pt_edwards.c',
+  'src/curves/curves.c',
+  'src/curves/ec_edwards.h',
+  'src/curves/ec_montgomery.h',
+  'src/curves/ec_params.h',
+  'src/curves/prj_pt.c',
+  )
+
+hash_mod_src = files(
+  'src/hash/hash_algs.c',
+  'src/hash/sm3.c',
+  'src/hash/streebog.c',
+  'src/hash/ripemd160.c',
+  'src/hash/belt-hash.c',
+  'src/hash/hmac.c',
+  'src/hash/bash224.c',
+  'src/hash/bash224.h',
+  'src/hash/bash256.c',
+  'src/hash/bash256.h',
+  'src/hash/bash384.c',
+  'src/hash/bash384.h',
+  'src/hash/bash512.c',
+  'src/hash/bash512.h',
+  'src/hash/bash.c',
+  'src/hash/bash.h',
+  'src/hash/sha224.c',
+  'src/hash/sha224.h',
+  'src/hash/sha256.c',
+  'src/hash/sha256.h',
+  'src/hash/sha2.h',
+  'src/hash/sha3-224.c',
+  'src/hash/sha3-224.h',
+  'src/hash/sha3-256.c',
+  'src/hash/sha3-256.h',
+  'src/hash/sha3-384.c',
+  'src/hash/sha3-384.h',
+  'src/hash/sha3-512.c',
+  'src/hash/sha3-512.h',
+  'src/hash/sha384.c',
+  'src/hash/sha384.h',
+  'src/hash/sha3.c',
+  'src/hash/sha3.h',
+  'src/hash/sha512-224.c',
+  'src/hash/sha512-224.h',
+  'src/hash/sha512-256.c',
+  'src/hash/sha512-256.h',
+  'src/hash/sha512.c',
+  'src/hash/sha512_core.c',
+  'src/hash/sha512_core.h',
+  'src/hash/sha512.h',
+  'src/hash/shake256.c',
+  'src/hash/shake256.h',
+  'src/hash/shake.c',
+  'src/hash/shake.h',
+  )
+
+sig_mod_src = files(
+  'src/sig/decdsa.c',
+  'src/sig/decdsa.h',
+  'src/sig/ecdsa.c',
+  'src/sig/ecdsa.h',
+  'src/sig/ecfsdsa.c',
+  'src/sig/ecfsdsa.h',
+  'src/sig/ecgdsa.c',
+  'src/sig/ecgdsa.h',
+  'src/sig/eckcdsa.c',
+  'src/sig/eckcdsa.h',
+  'src/sig/ecosdsa.c',
+  'src/sig/ecosdsa.h',
+  'src/sig/ecrdsa.c',
+  'src/sig/ecrdsa.h',
+  'src/sig/ecsdsa.c',
+  'src/sig/ecsdsa.h',
+  'src/sig/eddsa.c',
+  'src/sig/eddsa.h',
+  'src/sig/fuzzing_ecdsa.c',
+  'src/sig/fuzzing_ecdsa.h',
+  'src/sig/fuzzing_ecgdsa.c',
+  'src/sig/fuzzing_ecgdsa.h',
+  'src/sig/fuzzing_ecrdsa.c',
+  'src/sig/fuzzing_ecrdsa.h',
+  'src/sig/ecdsa_common.c',
+  'src/sig/ecsdsa_common.c',
+  'src/sig/sig_algs.c',
+  'src/sig/sm2.c',
+  'src/sig/bign_common.c',
+  'src/sig/bign.c',
+  'src/sig/dbign.c',
+  'src/sig/bip0340.c',
+  )
+
+ecdh_mod_src = files(
+  'src/ecdh/ecccdh.c',
+  'src/ecdh/ecccdh.h',
+  'src/ecdh/x25519_448.c',
+  'src/ecdh/x25519_448.h',
+  )
+
+
+if get_option('use_external_rand').enabled()
+rand_src = files(get_option('with_rand'))
+else
+rand_src = files()
+endif
+
+key_mod_src = files('src/sig/ec_key.c')
+
+libarith_lib = static_library('libarith',
+    sources: [ fp_mod_src, nn_mod_src, rand_src, utils_arith_src ],
+    include_directories: ecc_inc,
+    link_language: 'c',
```

Comment:
Since you only have one language, you don't need to override meson's default priority ordering of which language to use as the linker driver in a target with mixed language sources.

Reply:
Hello @eli-schwartz ! You're perfectly right, this is a bad habit of multi-langages projects. I'm gonna fix it :-)

Reply:
fixed